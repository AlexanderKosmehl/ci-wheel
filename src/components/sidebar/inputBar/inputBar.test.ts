import {
  describe, expect, it, vi,
} from 'vitest';
import generateInputBar from './inputBar';

describe('generateButton', () => {
  const newElementCallback = vi.fn((_newElement: string) => {});

  const inputBar = generateInputBar({
    newEntryCallback: newElementCallback,
  });
  const inputField = inputBar.querySelector<HTMLInputElement>('[data-test=sidebarInput]');
  const inputButton = inputBar.querySelector<HTMLButtonElement>('[data-test=sidebarInputButton]');

  it('generates component correctly', () => {
    expect(inputBar).toMatchSnapshot();
  });

  it('adds working callback', () => {
    if (!inputField || !inputButton) throw Error('Missing Inputs!');

    const testValue = 'test';

    inputField.value = testValue;
    inputField.dispatchEvent(new Event('keyup'));
    inputButton.click();

    expect(inputField.textContent).toBe('');
    expect(newElementCallback).toBeCalled();
    expect(newElementCallback).toBeCalledWith(testValue);
  });
});
