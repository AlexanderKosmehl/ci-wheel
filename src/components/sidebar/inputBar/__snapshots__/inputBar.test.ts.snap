// Vitest Snapshot v1

exports[`generateButton > generates component correctly 1`] = `
<div
  class="_container_fc15b0"
>
  <input
    class="_input_6ed02a"
    data-test="sidebarInput"
    placeholder="Neuer Eintrag"
    type="text"
  />
  <button
    class="_button_7c1f96 _addButton_fc15b0"
    data-test="sidebarInputButton"
    disabled=""
  >
    <img
      class="_icon_7c1f96"
      src="/src/assets/add-icon.svg"
    />
  </button>
</div>
`;
